### PR TITLE
Add preview argument to Client to easy switch api endpoint

### PR DESCRIPTION
**Pull request by rasmusjp**
If the Client is constructed with `preview: true` the preview endpoint is used.

```typescript
const client = new Client({
  projectAlias: 'your-project-alias',
  apiKey: 'your-api-key',
  language: 'iso-code',
  preview: true
})
```

### CODE REVIEW

File: etc/headless-client.api.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 ```ts
 
 import { AxiosResponse } from 'axios';
-import FormData from 'form-data';
+import { default as FormData_2 } from 'form-data';
```
Comment (sitereactor) [inline etc/headless-client.api.md:8]: Question: Why is this necessary? Why not just `import { default as FormData } from 'form-data';`
Mostly just curious why its necessary to do :)

File: samples/vue/src/client.ts [hunk 2]
```diff
@@ -2,7 +2,10 @@ import { Client } from '@umbraco/headless-client'
 
 const client = new Client({
   projectAlias: process.env.VUE_APP_UMBRACO__PROJECTALIAS!,
-  language: 'en-US'
+  apiKey: process.env.VUE_APP_UMBRACO__APIKEY,
+  language: 'en-US',
+  preview: true,
+  cdnProxyUrl: 'https://preview.rainbowsrock.net',
```
Comment (sitereactor) [inline samples/vue/src/client.ts:8]: I don't think we should leave anything pointing to our development environment in our public repos 🤔

File: etc/headless-client.api.md (hunk 1 above)
Comment (rasmusjp) [reply etc/headless-client.api.md:8]: It's the auto generated documentation not sure why it's changed, but I did update some depenencies recently so that might be why

File: samples/vue/src/client.ts (hunk 2 above)
Comment (rasmusjp) [reply samples/vue/src/client.ts]: Not sure why I missed that one, only used if for testing

Comment (rasmusjp) [reply samples/vue/src/client.ts]: It's removed now :)
